include/modem/nrf_modem_lib_trace.h

```diff
@@ -18,9 +18,11 @@
  *
  * Initializes the module and the trace transport medium.
  *
+ * @param trace_heap Heap to use for modem traces
+ *
  * @return Zero on success, non-zero otherwise.
  */
-int nrf_modem_lib_trace_init(void);
+int nrf_modem_lib_trace_init(struct k_heap *trace_heap);
```

Comment:
`zephyr.h` should be included in this file because of `struct k_heap`. Alternatively we could use `extern` to get the trace heap into `nrf_modem_lib_trace.c` from `nrf_modem_os.c` as @lemrey suggested. Not certain what is best.